### PR TITLE
Feature monitoring API at feature group level dict proposal

### PR DESCRIPTION
**Pull request by javierdlrm**
This PR adds/fixes/changes...
- please summarize your changes to the code 
- and make sure to include all changes to user-facing APIs

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [ ] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/feature_group.py [hunk 1]
```diff
@@ -974,6 +979,196 @@ def get_validation_history(
...
+            with_reference_window: Configuration of the reference window.
+            compare_on: Configuration of the statistics comparison.
+            alert_on: Configuration of the alert.
+            on_schedule: Configuration of the scheduler.
+            description: Description of the monitoring configuration.
+
+        # Returns:
+            `FeatureMonitoringConfig` Feature monitoring configuration
+        """
+        fm_config_engine = self._get_instance_feature_monitoring_config_engine()
```
Comment (vatj) [inline python/hsfs/feature_group.py:1149]: You can skip assigning it to a variable

File: python/hsfs/core/monitoring_window_config.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def __str__(self):
         return self.json()
 
     def __repr__(self):
-        return json.dumps(humps.decamelize(self.to_dict()), indent=2)
+        return f"MonitoringWindowConfig(type: {self._window_config_type})"
```
Comment (vatj) [inline python/hsfs/core/monitoring_window_config.py:70]: I don't think we should make it harder to introspect entities quickly, even for us when we need to do debugging

**Review by vatj — COMMENTED** [review]
I think you should put some more thinking in good defaults if we want to go forward with this UX. For the sake of clarity can you add a comment to method which are intended to be public when it stabilises

File: python/hsfs/core/monitoring_window_config.py (hunk 2 above)
Comment (javierdlrm) [reply python/hsfs/core/monitoring_window_config.py:70]: For debugging purposes, I agree. But for introspection I think it's better to have a specific .describe() method.

**Pull request closed without merging**